File: src/backend/app/models/roadmap.py
from datetime import datetime
from pydantic import BaseModel

from app.models.skill import Skill


class Resource(BaseModel):
    title: str
    url: str


class RoadmapStep(BaseModel):
    id: str
    title: str
    description: str
    estimatedHours: int
    resources: list[Resource] = []
    skills: list[Skill] = []


class Roadmap(BaseModel):
    id: str
    userId: str
    goal: str
    steps: list[RoadmapStep]
    createdAt: datetime

File: src/frontend/types/roadmap.ts
import type { Skill } from "./skill";

export interface RoadmapStep {
  id: string;
  title: string;
  description: string;
  estimatedHours: number;
  resources: { title: string; url: string }[];
  skills: Skill[];
}

export interface Roadmap {
  id: string;
  userId: string;
  goal: string;
  steps: RoadmapStep[];
  createdAt: string;
}
